kotlin/src/main/kotlin/org/axonframework/extensions/kotlin/CommandGatewayExtensions.kt

```diff
@@ -19,43 +19,28 @@
  */
 package org.axonframework.extensions.kotlin
 
-import org.axonframework.commandhandling.CommandCallback
 import org.axonframework.commandhandling.CommandMessage
 import org.axonframework.commandhandling.gateway.CommandGateway
 import org.axonframework.messaging.MetaData
 import java.util.concurrent.TimeUnit
 
 /**
- * Extension for command gateway.
+ * Callback-style send with dedicated on-success and on-error functions (defaults do nothing)
  */
-object CommandGatewayExtensions {
-    /**
-     * Callback-style send with dedicated on-success and on-error functions (defaults do nothing)
-     */
-    inline fun <reified C : Any, reified R : Any?> CommandGateway.send(
-            command: C,
-            crossinline onSuccess: (commandMessage: CommandMessage<out C>, result: R, metaData: MetaData) -> Unit = { _, _, _ -> },
-            crossinline onError: (commandMessage: CommandMessage<out C>, exception: Throwable, metaData: MetaData) -> Unit = { _, _, _ -> }
-    ) {
-        this.send(command, CommandCallback<C, R> { commandMessage, callBack ->
-            val metaData = callBack.metaData ?: MetaData.emptyInstance()
-            if (callBack.isExceptional) {
-                onError(commandMessage, callBack.exceptionResult(), metaData)
-            } else {
-                onSuccess(commandMessage, callBack.payload, metaData)
-            }
-        })
-    }
+fun <C : Any, R : Any?> CommandGateway.send(
+    command: C,
+    onSuccess: (commandMessage: CommandMessage<out C>, result: R, metaData: MetaData) -> Unit = { _, _, _ -> },
+    onError: (commandMessage: CommandMessage<out C>, exception: Throwable, metaData: MetaData) -> Unit = { _, _, _ -> }
+): Unit = this.send(command, CombiningCommandCallback<C, R>(onError, onSuccess))
 
-    /**
-     * Reified version of send and wait.
-     */
-    inline fun <reified R : Any?> CommandGateway.sendAndWaitWithResponse(command: Any) =
-            this.sendAndWait<R>(command)
+/**
+ * Reified version of [CommandGateway.sendAndWait].
+ */
+inline fun <reified R : Any?> CommandGateway.sendAndWaitWithResponse(command: Any): R =
```

Comment:
Curious: Is it custom to use a different method name for extension functions? In my mind, it feels slightly more intuitive to overload, although I am not certain whether that's an option at all.

Reply:
Thanks for the feedback @snowe2010, much appreciated.